src/mixins/validatable-http-message.js

```diff
@@ -0,0 +1,450 @@
+        }
+      });
+    }
+  }
+
+  validateHeaders() {
+    this.ensureValidationHeaders();
+    this.setHeadersRealType();
+    this.setHeadersExpectedType();
+    this.setHeadersValidator();
+    this.runHeadersValidator();
+  }
+
+  setHeadersRealType() {
+    this.ensureValidationHeaders();
+    const { headers } = this;
+    this.validation.headers.realType =
+      headers instanceof Object && !Array.isArray(headers)
+        ? 'application/vnd.apiary.http-headers+json'
+        : null;
+  }
+
+  setHeadersExpectedType() {
+    this.ensureValidationHeaders();
+    const expectedHeaders = this.expected.headers;
+    this.validation.headers.expectedType =
+      expectedHeaders instanceof Object && !Array.isArray(expectedHeaders)
+        ? 'application/vnd.apiary.http-headers+json'
+        : null;
+  }
+
+  setHeadersValidator() {
+    const expectsJson =
+      this.validation.headers.expectedType ===
+      'application/vnd.apiary.http-headers+json';
+
+    if (
+      this.validation.headers.realType ===
+        this.validation.headers.expectedType &&
+      expectsJson
+    ) {
+      return (this.validation.headers.validator = 'HeadersJsonExample');
+    } else {
+      this.validation.headers.validator = null;
+
+      if (!this.validation.headers.results) {
+        this.validation.headers.results = [];
+      }
+
+      const entry = {
+        message: `\
+No validator found for real data media type \
+"${JSON.stringify(this.validation.headers.realType)}" \
+and expected data media type \
+"${JSON.stringify(this.validation.headers.expectedType)}".\
+        `,
+        severity: 'error'
+      };
+
+      this.validation.headers.results.push(entry);
+    }
+  }
+
+  runHeadersValidator() {
+    // throw new Error JSON.stringify @validation.headers.validator, null, 2
+    let validator;
+
+    if (this.validation.headers.validator == null) {
+      this.validation.headers.rawData = null;
+    } else {
+      const ValidatorClass = validators[this.validation.headers.validator];
+      validator = new ValidatorClass(this.headers, this.expected.headers);
+      this.validation.headers.rawData = validator.validate();
+    }
+
+    if (!Array.isArray(this.validation.headers.results)) {
+      this.validation.headers.results = [];
+    }
+
+    if (this.validation.headers.rawData !== null) {
+      const results = validator.evaluateOutputToResults();
+      this.validation.headers.results = results.concat(
+        this.validation.headers.results
+      );
+    }
+  }
+
+  validateBody() {
+    this.validation.body = {};
+    this.validation.body.results = [];
+
+    this.setBodyRealType();
+    this.setBodyExpectedType();
+    this.setBodyValidator();
+    this.runBodyValidator();
+  }
+
+  setBodyRealType() {
+    this.validation.body.realType = null;
+    const bodyType = typeof this.body;
+
+    if (bodyType !== 'string') {
+      throw new Error(
+        `Expected HTTP body to be a String, but got: ${bodyType}`
```

Comment:
Well, it *is* Object then :) The usefulness of this message would be to scream on usage of anything but a string. Even if it may confuse people about null/object, but it will be evident `null` is not a string anyway.